src/Editor/GUI/Editors/EntityEditor.cpp

```diff
@@ -179,24 +182,42 @@ void EntityEditor::SetVisible(bool visible) {
     this->visible = visible;
 }
 
-void EntityEditor::AnimationEditor(Component::Animation* animation) {
+void EntityEditor::AnimationControllerEditor(Component::AnimationController* animationController) {
     ImGui::Indent();
-    if (ImGui::Button("Select model##Animation"))
-        ImGui::OpenPopup("Select model##Animation");
+    if (ImGui::Button("Select animation controller##Animation"))
+        ImGui::OpenPopup("Select animation controller##Animation");
 
-    if (ImGui::BeginPopup("Select model##Animation")) {
-        ImGui::Text("Models");
+    if (ImGui::BeginPopup("Select animation controller##Animation")) {
+        ImGui::Text("Animation controllers");
         ImGui::Separator();
-        
-        if (resourceSelector.Show(ResourceList::Resource::Type::MODEL)) {
-            if (animation->riggedModel != nullptr)
-                Managers().resourceManager->FreeModel(animation->riggedModel);
-            
-            animation->riggedModel = Managers().resourceManager->CreateModel(resourceSelector.GetSelectedResource().GetPath());
+
+        if (resourceSelector.Show(ResourceList::Resource::Type::ANIMATION_CONTROLLER)) {
+            if (animationController != nullptr)
+                Managers().resourceManager->FreeAnimationController(animationController->controller);
+
+            animationController->controller = Managers().resourceManager->CreateAnimationController(resourceSelector.GetSelectedResource().GetPath());
         }
 
         ImGui::EndPopup();
     }
+
+    if (ImGui::Button("Select skeleton##Skeleton"))
+        ImGui::OpenPopup("Select skeleton##Skeleton");
+
+    if (ImGui::BeginPopup("Select skeleton##Skeleton")) {
+        ImGui::Text("Skeletons");
+        ImGui::Separator();
+
+        if (resourceSelector.Show(ResourceList::Resource::Type::SKELETON)) {
+            if (animationController != nullptr)
+                Managers().resourceManager->FreeSkeleton(animationController->skeleton);
+
+            animationController->skeleton = Managers().resourceManager->CreateSkeleton(resourceSelector.GetSelectedResource().GetPath());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Either the condition 'animationController!=nullptr' is redundant or there is possible null pointer dereference: animationController. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AnullPointerRedundantCheck)